extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/config/HierarchicalOTelConnectionConfigInterceptor.java

```diff
@@ -46,7 +46,7 @@ public HierarchicalOTelConnectionConfigInterceptor() {
 
     @Override
     public Iterator<String> iterateNames(final ConfigSourceInterceptorContext context) {
-        Set<String> names = new HashSet<>();
+        Set<String> names = new LinkedHashSet<>();
```

Comment:
The LinkedHashmap maintains a predictable iteration order... Can't remember why it is used here anymore.

Reply:
I added it now, before it was HashSet ... but the order wasn't the problem., so I guess we can go back to plain HashSet ... ?

Reply:
yeah, let's go back. This has a performance penalty.

Reply:
A? With all 1M checks, this is the penalty? :-)
At least we know how they are ordered in the first place.
But yeah, np ... we can go back.